src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -118,6 +123,70 @@ public synchronized void close() throws IOException {
             return;
         }
 
+        try {
+            stopGracefully();
+        } finally {
+            stopForcefully();
+        }
+    }
+
+    private void stopGracefully() {
+        //TODO: Temporarily limiting graceful shutdown to only processes that are created using ProcessImplForWin32
+        if (!(process instanceof ProcessImplForWin32)) {
+            return;
+        }
+        int pid = process instanceof ProcessImplForWin32
+                ? ((ProcessImplForWin32) process).getPid() : Processes.newPidProcess(process).getPid();
+        boolean sentConsoleCtrlEvent = false;
+        synchronized (Kernel32.INSTANCE) {
+            Kernel32 k32 = Kernel32.INSTANCE;
+            // if the console is already attached, a second attach call will fail.
+            if (!k32.FreeConsole()) {
+                logger.error("FreeConsole error {}", k32.GetLastError());
+            }
+            if (!k32.AttachConsole(pid)) {
+                logger.error("AttachConsole error {}", k32.GetLastError());
+                // Console already attached so we cannot signal it
+                if (k32.GetLastError() == ERROR_ACCESS_DENIED) {
+                    logger.info("AttachConsole failed for PID: {}, calling FreeConsole on it", pid);
+                    // Oddly, calling FreeConsole seems to make subsequent calls work?
+                    if (!k32.FreeConsole()) {
+                        logger.error("FreeConsole error {}", k32.GetLastError());
+                    }
+                }
+                return;
+            }
+            Kernel32Ex k32Ex = Kernel32Ex.INSTANCE;
+            try {
+                if (!k32Ex.SetConsoleCtrlHandler(null, true)) {
+                    logger.error("SetConsoleCtrlHandler add error {}", k32.GetLastError());
+                    return;
+                }
+                if (k32.GenerateConsoleCtrlEvent(Wincon.CTRL_C_EVENT, 0)) {
+                    sentConsoleCtrlEvent = true;
```

Comment:
Try adding a slight sleep (1s to start) after generating the event. Other implementations have added a sleep "so that we don't kill ourselves", so maybe that is necessary.